src/cose_int.h

```diff
@@ -27,21 +82,80 @@ typedef struct CounterSign1 COSE_CounterSign1;
 #define _countof(x) (sizeof(x) / sizeof(x[0]))
 #endif
 
-typedef struct _COSE_KEY {
+#ifdef __cplusplus
+class COSE_KEY {
```

Comment:
I'd say that all internal source code is cpp, so no check needed, public headers should have either a C api or (better) both, modern cpp and ABI stable C api

Reply:
I did this rather than do the conversion on dumper from c to cpp.  Otherwise I agree withyou